third_party/kmod/eret_hvc_smc_module.c

```diff
@@ -0,0 +1,115 @@
+/* SPDX-License-Identifier: GPL-2.0 */
+/*
+ * Copyright (c) 2019 Google LLC
+ *
+ **/
+
+#include <linux/fs.h>
+#include <linux/init.h>
+#include <linux/kernel.h>
+#include <linux/kobject.h>
+#include <linux/module.h>
+#include <linux/printk.h>
+#include <linux/slab.h>
+#include <linux/sysfs.h>
+#include <linux/uaccess.h>
+
+MODULE_LICENSE("GPL");
+MODULE_AUTHOR("Google");
+MODULE_DESCRIPTION("");
+MODULE_VERSION("0.1");
+
+#ifndef __aarch64__
+#  error Unsupported CPU. ARM64 required.
+#endif
+
+// Provides an endpoint on
+//   /sys/kernel/eret_hvc_smc/address
+// to which userspace programs can send their addresses that will be
+// speculatively fetched after speculating over ERET, HVC and SMC instructions.
+// Currently should be accessible only by root, because there is no checking of
+// those addresses.
+
+static ssize_t address_store(struct kobject *kobj, struct kobj_attribute *attr,
+                             const char *buf, size_t length) {
+  size_t address;
+  int *memory, res = kstrtoul(buf, 0, &address);
```

Comment:
multiple declarations where one is a pointer and one is very easy to misread. Please split into different lines.

Reply:
Done.